server/text_generation_server/layers/linear.py

```diff
@@ -29,9 +34,63 @@ def forward(self, input: torch.Tensor) -> torch.Tensor:
         return F.linear(input, self.weight, self.bias)
 
 
+class FastLinearROCm(torch.nn.Module):
+    def __init__(
+        self,
+        weight,
+        bias,
+    ) -> None:
+        super().__init__()
+        self.weight = torch.nn.Parameter(weight)
+        if bias is not None:
+            self.bias = torch.nn.Parameter(bias)
+        else:
+            self.bias = None
+
+    @classmethod
+    def load(cls, config, prefix: str, weights, bias: bool):
+        weight = weights.get_tensor(f"{prefix}.weight")
+        if bias:
+            bias = weights.get_tensor(f"{prefix}.bias")
+        else:
+            bias = None
+        return cls(weight, bias)
+
+    def forward(self, inp: torch.Tensor) -> torch.Tensor:
+        weight = self.weight
+        bias = self.bias
+
+        if SYSTEM == "rocm" and inp.numel() // inp.size(-1) == 1:
+            batched = False
```

Comment:
Could be more than 3 technically no ?


```python
shape = input.shape
input = input.view(-1, shape[-1])
...
out.view(*shape[:-1], out.shape[0])
```

Is slightly better no ?

Try not mixing shape and size too (and for consistency, using shape everywhere ideally)